x-pack/plugins/security_solution/public/flyout/shared/components/preview_link.tsx

```diff
@@ -22,20 +23,26 @@ import { HOST_PREVIEW_BANNER } from '../../document_details/right/components/hos
 import { UserPreviewPanelKey } from '../../entity_details/user_right';
 import { USER_PREVIEW_BANNER } from '../../document_details/right/components/user_entity_overview';
 import { NetworkPanelKey, NETWORK_PREVIEW_BANNER } from '../../network_details';
+import { RulePreviewPanelKey, RULE_PREVIEW_BANNER } from '../../rule_details/right';
+
+const PREVIEW_FIELDS = [HOST_NAME_FIELD_NAME, USER_NAME_FIELD_NAME, SIGNAL_RULE_NAME_FIELD_NAME];
 
 // Helper function to check if the field has a preview link
 export const hasPreview = (field: string) =>
-  field === HOST_NAME_FIELD_NAME ||
-  field === USER_NAME_FIELD_NAME ||
-  getEcsField(field)?.type === IP_FIELD_TYPE;
+  PREVIEW_FIELDS.includes(field) || getEcsField(field)?.type === IP_FIELD_TYPE;
 
 interface PreviewParams {
   id: string;
   params: Record<string, unknown>;
 }
 
 // Helper get function to get the preview parameters
-const getPreviewParams = (value: string, field: string, scopeId: string): PreviewParams | null => {
+const getPreviewParams = (
+  value: string,
+  field: string,
+  scopeId: string,
+  ruleId?: string
```

Comment:
`ruleId` is optional here, but I feel like the unit tests aren't really verifying what the behavior should be if it is?